specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/environmentSpecification.json

```diff
@@ -0,0 +1,414 @@
+      "type": "string",
+      "description": "Environment Version identifier.",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "definitions": {
+    "AssetPath": {
+      "description": "Details of an AssetUri.",
+      "required": [
+        "path"
+      ],
+      "type": "object",
+      "properties": {
+        "path": {
+          "description": "The path of file/directory.",
+          "pattern": "\\w",
+          "type": "string"
+        },
+        "isDirectory": {
+          "description": "Whether the path defines a directory or a single file.",
+          "type": "boolean"
+        }
+      }
+    },
+    "EnvironmentSpecificationVersion": {
+      "description": "Data Transfer Object (DTO) for an Execution Environment Specification. Name and\r\nversion should default to the AzureML-Minimal environment version tested with the\r\ncurrent SDK client version as this is a main use-case. Only supported properties\r\nlisted in EnvironmentSpecificationProperties are allowed. For specification types\r\nthat support additional context, AssetPaths may contain a single Blob URI entry\r\npointing to a snapshot of that context.",
+      "type": "object",
+      "properties": {
+        "specificationType": {
+          "description": "Must be either a Curated Environment, which is the default, or one of the\r\nAzure ML supported community-standard specification file types. SetupFile must\r\nmatch the SpecificationType. For example \"Conda\" type must have a corresponding\r\nenvironment.yml file, \"Pip\" type must have requirements.txt or pipfile, \"Docker\"\r\ntype must have a Dockerfile and so on.",
+          "enum": [
+            "Curated",
+            "Conda",
+            "Python",
+            "Pipfile",
+            "Docker",
+            "DockerImage",
+            "Spark",
+            "R",
+            "VisualStudioDevContainer"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "EnvironmentSpecificationType",
+            "modelAsString": false
+          }
+        },
+        "description": {
+          "description": "The asset description text.",
+          "type": "string"
+        },
+        "datastoreId": {
+          "description": "The asset datastoreId",
+          "type": "string"
+        },
+        "tags": {
+          "description": "The asset tag dictionary.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
+          "description": "The asset property dictionary.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "assetPaths": {
+          "description": "Details of asset paths in the datastore.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AssetPath"
+          }
+        },
+        "linkedResourceIds": {
```

Comment:
What are the valid keys here?  How does a user of the API know?

Reply:
removed linkedResourcedIds